setools/terulequery.py

```diff
@@ -141,6 +144,13 @@ def results(self):
         self.log.debug("Boolean: {0.boolean!r}, eq: {0.boolean_equal}, "
                        "regex: {0.boolean_regex}".format(self))
 
+        if self.target and self.target_self:
+            raise ValueError("target and target_self both specified")
```

Comment:
I think there should be a new exception and it should probably subclass RuntimeError (and SEToolsException) specifically for signalling conflicting options.